src/Core/ManifestInterface.php

```diff
@@ -4,6 +4,6 @@
 
 interface ManifestInterface extends MetadataAwareInterface
 {
-    public static function name(): string;
+    public function name(): string;
```

Comment:
I thought this through, and I don't think this is a good idea: the whole point of `ManifestInterface::name()` method to be static - is to give user some understanding, that he should not use this method to return fully formatted name.

With this method being dynamic - user can inject env, project, app and other objects to his manifest class, only in order to create fully-qualified name for manifest. Thus - there is a big chance of total misuse of this method.

Not only that, but also `makeManifest` method in `NamesHelper` will not work (see comment in that class).

I think we should leave this method static.

However, I understand your concerns about some `name` methods being static, and some methods - dynamic.

I think we should rename this method to something more specific, but I didn't came up with something good yet.
Maybe `shortName`?

For this PR we may just live this method static with an old name, and later we'll rename it.